app/code/Magento/Bundle/Model/ResourceModel/Indexer/Price/DisabledProductOptionPriceModifier.php

```diff
@@ -121,7 +121,7 @@ private function getWebsiteIdsOfProduct(int $entityId): array
         foreach ($connection->fetchCol($select) as $websiteId) {
             $this->websiteIdsOfProduct[$entityId][] = (int)$websiteId;
         }
```

Comment:
Looks like the issue coming from not declaring the empty array for the entity id. I would suggest adding it.
```suggestion
        $this->websiteIdsOfProduct[$entityId] = [];
        foreach ($connection->fetchCol($select) as $websiteId) {
            $this->websiteIdsOfProduct[$entityId][] = (int)$websiteId;
        }
```